doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,385 @@
+When we create this new `wt` instance, it will obey the glomming rules as
+specified in [Session Management Spec]. It might end up glomming to another
+existing window at that elevation level, or possibly create its own window.
+
+Similarly, for a new elevated window, we can make sure to pass the `-s -1` arguments
+to `wt`. These parameters indicate that we definitely want this command to run in a
+new window, regardless of the current glomming settings.
+
+```
+wt -s -1 new-tab [args...]
+```
+
+However, creating a new **split pane** is a little trickier. Invoking the `wt
+split-pane [args...]` is straightforward enough.
+
+⚠ **TODO**: For discussion:
+
+If the current window doesn't have the same elevation level as the
+requested profile, do we always want to just create a new split? If the command
+ends up glomming to an existing window, does that even make sense? That invoking
+an elevated split in an unelevated window would end up splitting the elevated
+window? It's very possible that the user wanted a split in the tab they're
+currently in, in the unelevated window, but they don't want a split in the
+elevated window.
+
+What if there's not space in the elevated window to create the split (but there
+would be in the current window)? That would sure make it seem like nothing
+happened, silently.
+
+We could alternatively have cross-elevation splits default to always opening a
+new tab. That might mitigate some of the odd
+
+#### Configure the Terminal to _always_ run elevated
+
+`elevated` is a per-profile property, not a global property. If a user
+wants to always have all instances of the Terminal run elevated, they
+could set `"elevated": true` in their profile defaults. That would cause _all_
+profiles they launch to always spawn as elevated windows.
+
+#### `elevated` in Actions
+
+
+Additionally, we'll add the `elevated` property to the `NewTerminalArgs` used in
+the `newTab`, `splitPane`, and `newWindow` actions. This is similar to how other
+properties of profiles can be overriden at launch time. This will allow windows,
+tabs and panes to all be created specifically as elevated windows.
+
+We'll also add an iterable command for opening a profile in an
+elevated tab, with the following json:
+
+```jsonc
+{
+    // New elevated tab...
+    "name": { "key": "NewElevatedTabParentCommandName", "icon": "UAC-Shield.png" },
+    "commands": [
+        {
+            "iterateOn": "profiles",
+            "icon": "${profile.icon}",
+            "name": "${profile.name}",
+            "command": { "action": "newTab", "profile": "${profile.name}", "elevated": true }
+        }
+    ]
+},
+```
+
+## Implementation Details
+
+### Starting an elevated process from an unelevated process
+
+It seems that we're able to create an elevated process by passing the `"runas"`
+verb to
+[`ShellExecute`](https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutea).
+So we could use something like
+
+```c++
+ShellExecute(nullptr,

```

Comment:
What will be the OneCore behavior here? We'll have to likely cross that bridge one day.

Reply:
Oh that's a good question. I haven't the faintest clue how you elevate on OneCore, but I'm _sure_ there aren't any docs on that. Are there different users on OneCore? Different integrity levels?

Reply:
(I've added a section in "Potential issues")